packages/core/src/hosted-form/iframe-content/hosted-input-vaulting-handler.ts

```diff
@@ -0,0 +1,62 @@
+import { IframeEventPoster } from '../../common/iframe';
+import { StorefrontVaultingRequestSender } from '../../payment';
+import { HostedFieldVaultingRequestEvent } from '../hosted-field-events';
+
+import HostedInputAggregator from './hosted-input-aggregator';
+import { HostedInputEvent, HostedInputEventType } from './hosted-input-events';
+import HostedInputValidator from './hosted-input-validator';
+
+export default class HostedInputVaultingHandler {
+    constructor(
+        private _inputAggregator: HostedInputAggregator,
+        private _inputValidator: HostedInputValidator,
+        private _eventPoster: IframeEventPoster<HostedInputEvent>,
+        private _vaultingRequestSender: StorefrontVaultingRequestSender,
+    ) {}
+
+    handle: (event: HostedFieldVaultingRequestEvent) => Promise<void> = async (event) => {
+        const {
+            payload: { data, fields },
+        } = event;
+        const values = this._inputAggregator.getInputValues();
+        const results = await this._inputValidator.validate(values);
+
+        this._eventPoster.post({
+            type: HostedInputEventType.Validated,
+            payload: results,
+        });
+
+        if (!results.isValid) {
+            return this._eventPoster.post({
+                type: HostedInputEventType.VaultingFailed,
+            });
+        }
+
+        const { default_instrument, cardholder_name, ...billingAddress } = fields;
+
+        const [expiry_month, expiry_year] = values.cardExpiry ? values.cardExpiry.split('/') : [];
+
+        try {
+            await this._vaultingRequestSender.submitPaymentMethod(data, {
```

Comment:
Also, it seems we're missing the verification nonce?

Reply:
I believe this endpoint doesn't require verification nonce.

Reply:
So as far as I understand we need to modify endpoint and add nonce check?

Reply:
It's a former endpoint, but I'm not sure if we can add captcha to it. So yeah, we'll keep it in mind.
cc @bc-akravets